### ISSUE TITLE
Weird FAILED! when loading "0" value on a key

### ISSUE THREAD

**Issue by adomenech73** [TerryHowe/ansible-modules-hashivault#189]
I got a yaml file with the k/v I want to load to vault like:

```yaml
vault_kvs:
  "commons.httpClient.connectionManagerTimeout": "11000"
  "commons.httpClient.socketTimeout": "60000"
  "commons.httpConnectionManager.connectionTimeout": "10000"
  "commons.httpConnectionManager.linger": "0"
  "commons.httpConnectionManager.maxConnections": "16"
  "commons.httpConnectionManager.maxConnectionsPerHost": "16"
```
Then I have a role that loads this data like:

```yaml
- name: Upload vault-kv from yaml file
  hashivault_write:
    url: "http://127.0.0.1:8200"
    authtype: "approle"
    role_id: "{{rundeck_role_id}}"
    secret: "/kv/config/{{app_name}}"
    verify: false
    data:
        "{{ vault_kvs }}"
  run_once: True
  delegate_to: "{{vault_server}}"
```

The problem comes with `"commons.httpConnectionManager.linger": "0"`line, meaning that:

- With this setup, role execution fails on the k/v upload with VALUE_SPECIFIED_IN_NO_LOG_PARAMETER
- If I comment this specific line the k/v upload works as expected
- If I change the value from 0 to 1 the k/v upload works as expected

**Comment by TerryHowe** [comment]
Related https://github.com/ansible/ansible/issues/16125

**Comment by TerryHowe** [comment]
Fixed in 4.4.1
